app/lib/features/home/views/home_body.dart

```diff
@@ -0,0 +1,528 @@
+import 'package:acter/common/providers/keyboard_visbility_provider.dart';
+import 'package:acter/common/themes/app_theme.dart';
+import 'package:acter/common/utils/constants.dart';
+import 'package:acter/common/utils/routes.dart';
+import 'package:acter/features/cross_signing/providers/notifiers/verification_notifiers.dart';
+import 'package:acter/features/cross_signing/providers/verification_providers.dart';
+import 'package:acter/features/cross_signing/views/request_cancelled.dart';
+import 'package:acter/features/cross_signing/views/request_created.dart';
+import 'package:acter/features/cross_signing/views/request_done.dart';
+import 'package:acter/features/cross_signing/views/request_ready.dart';
+import 'package:acter/features/cross_signing/views/sas_accepted.dart';
+import 'package:acter/features/cross_signing/views/sas_cancelled.dart';
+import 'package:acter/features/cross_signing/views/sas_keys_exchanged.dart';
+import 'package:acter/features/cross_signing/views/sas_started.dart';
+import 'package:acter/features/cross_signing/views/verification_request.dart';
+import 'package:acter/features/home/providers/client_providers.dart';
+import 'package:acter/features/home/providers/navigation.dart';
+import 'package:acter/features/home/widgets/sidebar_widget.dart';
+import 'package:acter_flutter_sdk/acter_flutter_sdk_ffi.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_adaptive_scaffold/flutter_adaptive_scaffold.dart';
+import 'package:flutter_keyboard_visibility/flutter_keyboard_visibility.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+import 'package:logging/logging.dart';
+import 'package:screenshot/screenshot.dart';
+
+final _log = Logger('a3::home::body');
+
+const homeShellKey = Key('home-shell');
+ScreenshotController screenshotController = ScreenshotController();
+
+class HomeBody extends ConsumerStatefulWidget {
+  final StatefulNavigationShell navigationShell;
+  final bool hasFirstSynced;
+
+  const HomeBody({
+    super.key = homeShellKey,
+    required this.navigationShell,
+    required this.hasFirstSynced,
+  });
+
+  @override
+  ConsumerState<ConsumerStatefulWidget> createState() => HomeBodyState();
+}
+
+class HomeBodyState extends ConsumerState<HomeBody> {
+  final GlobalKey<ScaffoldState> _key = GlobalKey<ScaffoldState>();
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    ref.listen(verificationStateProvider, onStateChange);
+    final bottomBarNav = ref.watch(bottomBarNavProvider(context));
+    final keyboardVisibility = ref.watch(keyboardVisibleProvider);
+    return CallbackShortcuts(
```

Comment:
I just assumed that all that below here basically didn't change from the place it was copied from ...